app/controllers/moderations_controller.rb

```diff
@@ -42,6 +42,12 @@ def actions_panel
     has_no_relevant_adjustments = @adjustments.pluck(:tag_id).intersection(tag_mod_tag_ids).size.zero?
     can_be_adjusted = @moderatable.tags.ids.intersection(tag_mod_tag_ids).size.positive?
 
+    @is_user_flagged = Reaction.user_vomits.where(
```

Comment:
For non-admin moderators, their flags are given the status `valid` until an admin confirms them; then the status changes to `confirmed`.

The flags of admin-level moderators are automatically confirmed.

So if we query just for "status: confirmed" reactions, the "Flag/Unflag User" toggle will only work for admin-level moderators, but not the others. This is the behavior I am currently experiencing in the app.

I suggest this:
1. Add a scope to `reaction.rb` that queries for "valid" or "confirmed" statuses:
```ruby
scope :valid_or_confirmed, -> { where(status: %w[valid confirmed]) }
```

2. When defining `@is_user_flagged`, include this scope:
```ruby
Reaction.user_vomits.valid_or_confirmed.where(user_id: ......)
```

3. Be sure to apply the above change to wherever you defined `@is_user_flagged` for the purpose of the Actions Panel. Bear in mind that not every instance of `@is_user_flagged1 should be defined as above.

Please let me know if the above does not make sense. 🙏🏾